src/librustc_typeck/check/mod.rs

```diff
@@ -5023,14 +5036,49 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                 path_segs.push(PathSeg(generics_def_id, last));
             }
 
-            // Case 2. Reference to a top-level value.
+            // Case 2. Reference to a variant constructor.
+            Def::VariantCtor(def_id, ..) => {
+                if self.tcx.features().type_alias_enum_variants {
```

Comment:
This condition needs to be removed (feature gates shouldn't change behavior, as we previously discussed).